google/cloud/spanner/client.h

```diff
@@ -371,6 +373,42 @@ class Client {
 std::shared_ptr<Connection> MakeConnection(
     Database const& db, ConnectionOptions const& options = ConnectionOptions());
 
+namespace internal {
+/**
+ * Execute a function in the context of a read-write transaction, with
+ * automatic retries if the transaction commit results in an abort.
+ *
+ * The caller-provided function will be passed the `Client` argument and a
+ * newly created read-write `Transaction`. It should use these objects to
+ * issue any `Read()`s, `ExecuteSql()`s, etc., and return the `Mutation`s to
+ * commit, or an error (which causes the transaction to be rolled back).
+ *
+ * The lock priority of the transaction increases after each prior aborted
+ * transaction, meaning that the next attempt has a slightly better chance
+ * of success than before.
+ *
+ * @param client how to contact Cloud Spanner
+ * @param opts options for the transaction created by this function
+ * @param f the function to call in the transaction.
+ * @param retry_policy controls for how long (or how many times) will the
+ *     function retry the operation when there is a retryable failure.
```

Comment:
Just to reiterate my opinion ... I don't think `RunTransaction()` should be retrying anything but `ABORTED`.  Everything else is the responsibility of whatever retries normal (non-`RunTransaction()`) calls.

Reply:
I think I agree with that, although this PR doesn't change that behavior AFAICT?
Can you open an issue?

Reply:
Done.  #529